test/integration/preload-viewport/test/index.test.js

```diff
@@ -436,18 +436,23 @@ describe('Prefetching Links in viewport', () => {
   it('should not add an another observer for a prefetched page', async () => {
     // info: both `/` and `/de-duped` ref the `/first` page, which we don't
     // want to be re-fetched/re-observed.
-    const browser = await webdriver(appPort, '/')
-    await browser.eval(`(function() {
-      window.calledPrefetch = false
-      window.next.router.prefetch = function() {
-        window.calledPrefetch = true
-        return Promise.resolve()
-      }
-      window.next.router.push('/de-duped')
-    })()`)
-    await waitFor(2 * 1000)
-    const calledPrefetch = await browser.eval(`window.calledPrefetch`)
-    expect(calledPrefetch).toBe(false)
+    const browser = await webdriver(appPort, '/', {
+      waitHydration: false,
+    })
+
+    let requests = []
+
+    browser.on('request', (req) => {
```

Comment:
This doesn't seem to be testing the same thing it was previously as we were previously ensuring the observer wasn't being re-added after a prefetch occurred

Reply:
the title confused me a bit as well (and we probably ought to change it), if that was the intention at the start the next/link implementation is definitely not doing that anymore, as we're definitely attaching an intersection observer.